vpr/src/place/cut_spreader.cpp

```diff
@@ -193,9 +193,17 @@ int CutSpreader::tiles_at(int x, int y) {
  * * grow merged to include all mergee grids
  */
 void CutSpreader::merge_regions(SpreaderRegion& merged, SpreaderRegion& mergee) {
+    size_t max_x = g_vpr_ctx.device().grid.width();
+    size_t max_y = g_vpr_ctx.device().grid.height();
     for (int x = mergee.bb.xmin(); x <= mergee.bb.xmax(); x++)
         for (int y = mergee.bb.ymin(); y <= mergee.bb.ymax(); y++) {
-            VTR_ASSERT(reg_id_at_grid[x][y] == mergee.id);
+            if (x >= max_x || x < 0 || y >= max_y || y < 0) { //location is not within the chip
```

Comment:
This sequence of (1) compute max_x, max_y and (2) check if a location is in the grid comes up a few times. Code would be shorter if you made a helper function loc_is_in_grid (x. y).  If you're worried it's performance critical, you can write it in the header file and make it inline (then it won't really result in a function call, but the code will be more readable).
I don't think we have a function for that, but again worth a quick check of the code to see.